### ISSUE TITLE
Incorrect use of filter function

### ISSUE THREAD

**Issue by brianmay** [jazzband/django-pipeline#676]
https://github.com/jazzband/django-pipeline/blob/5ec680174d1897498ec9c6652a4130bc0d39dddd/pipeline/compilers/__init__.py#L119

This line has:

```
argument_list = filter(None, argument_list)
```

But filter doesn't return a list, it returns a filter object that can only be iterated once. This in turn means that the errors do not include the correct value:

```
pipeline.exceptions.CompilerError: <filter object at 0x7f9d1073ab38> exit code 126
```

I believe this line should read be:

```
argument_list = list(filter(None, argument_list))
```

**Comment by nasirhjafri** [comment]
Yep you are right, It should be `list(filter(None, argument_list))`
